presentation/src/main/java/me/androidbox/presentation/agenda/viewmodel/AgendaViewModel.kt

```diff
@@ -62,19 +62,30 @@ class AgendaViewModel @Inject constructor(
             .minusSeconds(1L)
     }
 
-    fun fetchAgendaItems(agendaDate: ZonedDateTime = ZonedDateTime.now(ZoneId.systemDefault())) {
+    fun fetchAgendaItems(agendaDate: ZonedDateTime = ZonedDateTime.now(ZoneId.systemDefault()), isSwipeToRefresh: Boolean = false) {
         agendaJob?.cancel()
 
         agendaJob = viewModelScope.launch {
             agendaLocalRepository.fetchAgenda(getStartOffCurrentDay(agendaDate).toEpochSecond(), getEndOfCurrentDay(agendaDate).toEpochSecond())
                 .collectLatest { responseState ->
                     when(responseState) {
                         ResponseState.Loading -> {
-                            /* TODO Show some form of loading */
+                            if(isSwipeToRefresh) {
+                                _agendaScreenState.update { agendaScreenState ->
+                                    agendaScreenState.copy(
+                                        isRefreshingAgenda = true
+                                    )
+                                }
+                            }
```

Comment:
Flag is check here and if its true we are doing a swipe to refresh so update the state and show the swipe progress indicator